### PR TITLE
`azurerm_hdinsight_interactive_query_cluster`: Removing unsupported acc test

### PR DESCRIPTION
**Pull request by jiaweitao001**
* According to [official doc](https://learn.microsoft.com/en-us/azure/hdinsight/hdinsight-autoscale-clusters#cluster-compatibility), hdinsight interactive Query clusters can only be configured for schedule-based scaling, not load-based.

* Tests are passing.
```
=== RUN   TestAccAzureRMHDInsightInteractiveQueryCluster_autoscale
=== PAUSE TestAccAzureRMHDInsightInteractiveQueryCluster_autoscale
=== CONT  TestAccAzureRMHDInsightInteractiveQueryCluster_autoscale
--- PASS: TestAccAzureRMHDInsightInteractiveQueryCluster_autoscale (1315.77s)
PASS

```

### CODE REVIEW

File: internal/services/hdinsight/hdinsight_interactive_query_cluster_resource.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ var hdInsightInteractiveQueryClusterWorkerNodeDefinition = HDInsightNodeDefiniti
 	CanSpecifyInstanceCount: true,
 	MinInstanceCount:        1,
 	CanSpecifyDisks:         false,
-	CanAutoScaleByCapacity:  true,
+	CanAutoScaleByCapacity:  false,
```
Comment (mbfrahry) [inline internal/services/hdinsight/hdinsight_interactive_query_cluster_resource.go:35]: This is a breaking change. This used to work which means some people may still have configuration built on it. We should deprecate this instead

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @jiaweitao001, it looks like what you've submitted is a breaking change and we should deprecate it for 4.0 instead

File: internal/services/hdinsight/hdinsight_interactive_query_cluster_resource.go (hunk 1 above)
Comment (jiaweitao001) [reply internal/services/hdinsight/hdinsight_interactive_query_cluster_resource.go]: Updated to deprecate in 4.0.

**Review by mbfrahry — CHANGES_REQUESTED** [review]
This [block](https://github.com/hashicorp/terraform-provider-azurerm/pull/21981/files#diff-7eeb781709dd8b07571c2ab69c7a8c8136e163e30027eae4b74ace3b4d266bc9R118) can be removed as you're adding/modifying it down below

**Review by mbfrahry — APPROVED** [review]
LGTM! Thanks @jiaweitao001

**Pull request merged**